docs/developing_iree/profiling.md

```diff
@@ -0,0 +1,217 @@
+# Profiling
```

Comment:
Nice docs! How about including some images in addition to the text?
You could add images for GUI configuration like `Fill out the "Application Executable" and "Command Line Arguments"` and/or for a glance at what each tool offers (e.g. screenshot of a trace for an interesting workload)

Reply:
SG.  Added a TODO and will update with some pictures later.